File: 189-rotate-array/189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        if(len(nums)==1):
            return nums
        if(k>len(nums)):
            k=k%len(nums)
        x=nums[-k:]+nums[:-k]
        nums.clear()
        for i in x:
            nums.append(i)
        return nums
        

File: 189-rotate-array/189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& v, int k) {
        if(v.size()==1)return;
        if(k>v.size())k=k%v.size();
        std::rotate(v.begin(),v.begin()+v.size()-k,v.end());
    }
};